github-pullrequest-plugin/src/main/java/org/jenkinsci/plugins/github/pullrequest/trigger/check/PullRequestToCauseConverter.java

```diff
@@ -44,10 +41,15 @@ public static PullRequestToCauseConverter toGitHubPRCause(GitHubPRRepository loc
 
     @Override
     public GitHubPRCause apply(final GHPullRequest remotePR) {
-        return from(trigger.getEvents())
-                .transform(toCause(remotePR))
-                .filter(notNull())
-                .firstMatch(new SkippedCauseFilter(listener)).orNull();
+        for (GitHubPREvent event: trigger.getEvents()) {
```

Comment:
That would contradict to class name. What you want achieve is more PullRequestToRunCause